node/channel.js

```diff
@@ -493,7 +511,11 @@ TChannel.prototype.register = function register(name, options, handler) {
 TChannel.prototype.address = function address() {
     var self = this;
     if (self.serverSocket) {
-        return self.serverSocket.address() || null;
+        if (self.listenFd !== null && self.listenFd !== undefined) {
+            return { port: self.port, family: 'IPv4', address: self.address };
```

Comment:
self.address is a method. Might mean `host`. family is not necessarily IPv4. It might be best to carry the address return value over from a prior listen.